### PR TITLE
[SPARK-38620][WEBUI] Replace `value.formatted(formatString)` with `formatString.format(value)` to clean up compilation warning

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
There are some compilation warning as follows:

```
[WARNING] /spark-source/streaming/src/main/scala/org/apache/spark/streaming/ui/StreamingPage.scala:67: [deprecation @ org.apache.spark.streaming.ui.RecordRateUIData.formattedAvg.$anonfun | origin=scala.Predef.StringFormat.formatted | version=2.12.16] method formatted in class StringFormat is deprecated (since 2.12.16): Use `formatString.format(value)` instead of `value.formatted(formatString)`,
or use the `f""` string interpolator. In Java 15 and later, `formatted` resolves to the new method in String which has reversed parameters.
[WARNING] /spark-source/sql/core/src/main/scala/org/apache/spark/sql/streaming/ui/StreamingQueryPage.scala:201: [deprecation @ org.apache.spark.sql.streaming.ui.StreamingQueryPagedTable.row | origin=scala.Predef.StringFormat.formatted | version=2.12.16] method formatted in class StringFormat is deprecated (since 2.12.16): Use `formatString.format(value)` instead of `value.formatted(formatString)`,
or use the `f""` string interpolator. In Java 15 and later, `formatted` resolves to the new method in String which has reversed parameters.
[WARNING] /spark-source/sql/core/src/main/scala/org/apache/spark/sql/streaming/ui/StreamingQueryPage.scala:202: [deprecation @ org.apache.spark.sql.streaming.ui.StreamingQueryPagedTable.row | origin=scala.Predef.StringFormat.formatted | version=2.12.16] method formatted in class StringFormat is deprecated (since 2.12.16): Use `formatString.format(value)` instead of `value.formatted(formatString)`,
or use the `f""` string interpolator. In Java 15 and later, `formatted` resolves to the new method in String which has reversed parameters.
```

this pr change to use `formatString.format(value)` to clean up these warnings.

### Why are the changes needed?
Clean up compilation warning.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA

### CODE REVIEW

File: streaming/src/main/scala/org/apache/spark/streaming/ui/StreamingPage.scala [hunk 1]
```diff
@@ -64,7 +64,7 @@ private[ui] class RecordRateUIData(val data: Seq[(Long, Double)]) {
 
   val avg: Option[Double] = if (data.isEmpty) None else Some(data.map(_._2).sum / data.size)
 
-  val formattedAvg: String = avg.map(_.formatted("%.2f")).getOrElse("-")
+  val formattedAvg: String = avg.map("%.2f".format(_)).getOrElse("-")
```
[the remark this answers is not in the document]
Comment (LuciferYang) [reply streaming/src/main/scala/org/apache/spark/streaming/ui/StreamingPage.scala:67]: can't use `.map("%.2f".format)`, compile error as follows:

```
spark/streaming/src/main/scala/org/apache/spark/streaming/ui/StreamingPage.scala:67: type mismatch;
 found   : Seq[Any] => String
 required: Double => ?

```

[the remark this answers is not in the document]
Comment (srowen) [reply streaming/src/main/scala/org/apache/spark/streaming/ui/StreamingPage.scala:67]: Ah OK nevermind

**Comment by srowen** [comment]
Merged to master

**Comment by LuciferYang** [comment]
thanks @srowen

**Pull request closed without merging**